src/NServiceBus.Transport.RabbitMQ/Sending/BasicPropertiesExtensions.cs

```diff
@@ -57,7 +57,11 @@ public static void Fill(this IBasicProperties properties, OutgoingMessage messag
                 }
             }
 
-            if (messageHeaders.TryGetValue(NServiceBus.Headers.ContentType, out var contentType) && contentType != null)
+            if (dispatchProperties.TryGetValue(NativeIntegrationDispatchPropertiesExtensions.ContentTypeAttribute, out var contentType) && contentType != null)
```

Comment:
By looking at the PR description on the order of evaluation, it is unclear why dispatch properties should take precedence over the presence of the header. By looking at the code change one could argue that the now content type attribute is a temporary header (one that gets persisted into the outbox too) which the only difference that is never in the "NSB headers" but technically for integration scenarios in a raw dispatch setting the `NServiceBus.Headers.ContentType` would achieve the same and it is not clear why that route is not desirable.